src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -495,6 +497,23 @@ async Task<MsalAccessTokenCacheItem> ITokenCacheInternal.FindAccessTokenAsync(
             return msalAccessTokenCacheItem;
         }
 
+        private void FilterTokensByAdditionalKeyComponents(List<MsalAccessTokenCacheItem> accessTokens, AuthenticationRequestParameters requestParams)
+        {
+            if (requestParams.CacheKeyComponents != null)
+            {
+                accessTokens.FilterWithLogging(item =>
+                    item.AdditionalCacheKeyComponents != null &&
+                    CollectionHelpers.AreDictionariesEqual(item.AdditionalCacheKeyComponents, requestParams.CacheKeyComponents),
```

Comment:
did we consider benchmarking this?

Reply:
@gladjohn please see https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/5107#discussion_r1942270849